composeApp/src/androidMain/kotlin/io/github/couchtracker/UserContext.kt

```diff
@@ -0,0 +1,51 @@
+package io.github.couchtracker
+
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.staticCompositionLocalOf
+import io.github.couchtracker.db.app.AppData
+import io.github.couchtracker.db.app.User
+import io.github.couchtracker.db.app.UserManager
+import org.koin.compose.koinInject
+
+val LocalUserManagerContext = staticCompositionLocalOf<UserManager> { error("no default user context") }
+
+@Composable
+fun UserContext(content: @Composable () -> Unit) {
+    val appDb = koinInject<AppData>()
+
+    val usersState = remember { appDb.userQueries.selectAll() }.asListState()
+    val users = usersState.value
+
+    LaunchedEffect(users) {
+        if (users?.isEmpty() == true) {
+            appDb.userQueries.insert(
+                name = "Default",
+                externalFileUri = null,
+            )
+        }
+    }
+
+    if (users.isNullOrEmpty()) {
+        Text("Loading...") // TODO do better
+    } else {
+        UserContext(users = users, content = content)
+    }
+}
+
+@Composable
+private fun UserContext(users: List<User>, content: @Composable () -> Unit) {
+    val coroutineScope = rememberCoroutineScope()
+    val userManager = remember { UserManager(users, coroutineScope) }
+    LaunchedEffect(users) {
+        userManager.updateListOfUsers(users)
+    }
```

Comment:
I'm not necessarily saying it's better, but if you want to avoid the dance between Compose and coroutine, this is an option:

```
    val userManagerFlow = remember {
        appDb.userQueries
            .selectAll()
            .asFlow()
            .mapToList(Dispatchers.IO)
            .runningFold(null as UserManager?) { userManager, users ->
                if (userManager == null) {
                    UserManager(users, coroutineScope)
                } else {
                    userManager.also { it.updateListOfUsers(users) }
                }
            }
    }
    val userManager by userManagerFlow.collectAsStateWithLifecycle(null)
    if (userManager == null) {
        ...
    } else {
        ...
    }

```

Reply:
I think the current version is easier and more readable.